DHaven.Faux/Compiler/TypeFactory.cs

```diff
@@ -40,37 +38,23 @@ static TypeFactory()
         internal static TService CreateInstance<TService>(string className)
             where TService : class // really interface
         {
+            var typeInfo = typeof(TService);
             EnsureAssemblyIsGenerated();
-            
+
             var type = generatedAssembly?.GetType(className);
             var constructor = type?.GetConstructor(EmptyTypes);
             return constructor?.Invoke(EmptyParams) as TService;
         }
-                
+
         private static void EnsureAssemblyIsGenerated()
         {
-            // Delay until all services have been registered first.
-            if (generatedAssembly != null)
```

Comment:
The assembly needs to be created once, and this pattern needs to remain (i.e. the double-check locked version)